src/components/page-donate/page-donate.tsx

```diff
@@ -19,54 +8,50 @@ export class PageDonate {
   @State() private amount?: number | null;
   @State() private showConfirmation: boolean = false;
   @State() private canNativeShare: boolean = false;
+  @State() private referral?: string | null;
 
   public componentWillLoad() {
     document.title = `Donate | Pizza to the Polls`;
-  }
+    const urlParams = new URLSearchParams(window.location.search);
+    this.referral = urlParams.get("referral") || "";
 
-  public render() {
-    let handler: any = null;
-    const StripeCheckout: any = (window as any).StripeCheckout;
+    if (urlParams.get("success") && urlParams.get("amount_usd")) {
+      this.amount = parseInt(urlParams.get("amount_usd") as string, 10);
+      this.showConfirmation = true;
+    }
+  }
 
-    // Get referral from URL
-    const urlParams = new URLSearchParams(window.location.search);
-    const referral = urlParams.get("referral") || "";
+  public async donate(amount: number) {
+    const resp = await fetch(`${process.env.PIZZA_BASE_DOMAIN}/donations`, {
+      body: JSON.stringify({ amountUsd: amount, referrer: this.referral }),
+      method: "POST",
+      mode: "cors",
+      headers: { "Content-Type": "application/json" },
+    });
```

Comment:
IIUC stripe will send a charge webhook to the donationsController with https://dashboard.stripe.com/webhooks @mojowen -- thoughts on having polls.pizza sending the ping vs getting it from Stripe? I don't think it's a big deal since the Donation entity has a unique check on the stripe charge id. Though it may double the hits to the donationController.

Reply:
Nvm, didn't see the recent changes to DonationsController.

Reply:
Yep--just for the record, this simply initiates the Stripe checkout. When the checkout succeeds, it looks like the plan is for Stripe to ping a separate endpoint on the Donations controller to register that payment with us.